Add OPW 2.19 changelog to docs (#38154)
DataDog/documentation: content/en/observability_pipelines/guide/upgrade_worker.md

This guide goes over how to upgrade to a specific Worker version and the updates for that version.

## Worker version 2.19.0

To upgrade to Worker version 2.19.0:

- Docker: Run the `docker pull` command for the [2.19.0 image][49].
- Kubernetes: See the [Helm chart][2] and [Upgrade the Worker][37].
- APT: Run the command `apt-get install observability-pipelines-worker=2.19.0`.
- RPM: Run the command `sudo yum install observability-pipelines-worker-2.19.0`.

Worker version 2.19.0 gives you access to the following:

#### Enhancements

> - A `server_name` TLS option has been added to the BYOC Logs, HTTP Client, Socket, and Syslog destinations and to the HTTP Client source. It overrides the SNI and certificate hostname used for the TLS handshake, which is applicable when the dialed address does not match the certificate's Common Name or Subject Alternative Name.

Reviewer: ```suggestion
- A `server_name` TLS option has been added to the BYOC Logs, HTTP Client, Socket, and Syslog destinations, and to the HTTP Client source. It overrides the SNI and certificate hostname used for the TLS handshake, which is applicable when the dialed address does not match the certificate's Common Name or Subject Alternative Name.
```

- The Splunk TCP source now supports hot-reloading TLS certificates, so certificate rotations are applied without restarting the Worker.
- The BYOC Logs destination now supports hot-reloading mTLS files.
- The Worker now supports a configurable graceful shutdown limit.
- The Amazon S3 destination now supports SSE-KMS encryption using the `server_side_encryption` and `ssekms_key_id` options.

#### Fixes

- Fixed an issue in the Reduce processor where a timestamp field whose name requires quoting in a path (for example, `"created.at"` or `"event-time"`) had its `_end` companion field silently dropped from the reduced event. The companion field is now placed correctly next to the base field.
- Fixed a configuration reload issue. If a reload changed a component's type while keeping the same name, such as replacing a source named `X` with a processor named `X`, any downstream processor or destination still reading from `X` now reconnects to the new component.

> - The Syslog source (TCP mode) has been fixed to handle over-length, length-prefixed message split across multiple reads.

Reviewer: ```suggestion
- The Syslog source (TCP mode) has been fixed to handle an over-length, length-prefixed message split across multiple reads.
```

> - Fixed an issue in the Syslog source (TCP mode) to handle over-length, length-prefixed message split across multiple reads.

Reviewer: ```suggestion
- The Syslog source (TCP mode) has been fixed to handle over-length, length-prefixed message split across multiple reads.
```

> - The Logstash source has been fixed to acknowledge only completed windows instead of sometimes sending a partial acknowledgment.

Reviewer: @mikhacim just some edits along the same lines as the other Logstash fix we updated.

```suggestion
- The Logstash source has been fixed to only send an acknowledgment when all events of a batch are received instead of sending a partial acknowledgment.
```

> - Fixed an issue in the Logstash source to acknowledge only completed windows instead of sometimes sending a partial acknowledgment. Partial acknowledgments could confuse proxies that expect a single acknowledgment per window and cause errors on subsequent batches.

Reviewer: What is the "completed windows" referring to? Also is it necessary to add the issue with partial acknowledgements since its been fixed?

```suggestion
- The Logstash source has been fixed to acknowledge only completed windows instead of sometimes sending a partial acknowledgment.
```

> - The Logstash source now rejects a window-size frame that arrives before the current window has received all of its advertised events, closing the connection with a fatal decode error instead of continuing.

Reviewer: What is the "window-size frame" referring to?

Author: It's a logstash-specific term, the bug case is that a sender that sends a new window-size frame before the current window's N events have all arrived, i.e. it starts announcing the next batch mid-batch, while events 1->n of the current window are still incomplete.

Instead, we are saying the current window's events must be received, otherwise we will reject new events from the next window.

Author: maybe advertised isn't the right term

Reviewer: What about using "batch of events"?

```suggestion
- The Logstash source now rejects a batch of events that arrives before all events in the current batch has been received, closing the connection with a fatal decode error instead of continuing.
```

> - Fixed the Syslog codec silently ignoring short-form severity keywords (`crit`, `emerg`, `err`, `info`, `warn`) and defaulting to `informational`. Both short-form and full-form severity names are now accepted.

Reviewer: ```suggestion
- Fixed the Syslog codec that was silently ignoring short-form severity keywords (`crit`, `emerg`, `err`, `info`, `warn`) and defaulting to `informational`. Both short-form and full-form severity names are now accepted.
```

> - Fixed an issue in the Custom Processor functions `parse_key_value`, `parse_cef`, `decode_mime_q`, and `parse_ruby_hash` on inputs with lines of 65,535 bytes or more.

Reviewer: Okay, just a small suggestion then

```suggestion
- Fixed an issue in the Custom Processor functions `parse_key_value`, `parse_cef`, `decode_mime_q`, and `parse_ruby_hash` that had inputs with lines of 65,535 bytes or more.
```

> - Fixed an issue in the Custom Processor functions `parse_key_value`, `parse_cef`, `decode_mime_q`, and `parse_ruby_hash` on inputs with lines of 65,535 bytes or more.

Reviewer: What was the issue?

Author: For which part?

Reviewer: This one:

> Fixed an issue in the Custom Processor functions `parse_key_value`, `parse_cef`, `decode_mime_q`, and `parse_ruby_hash` on inputs with lines of 65,535 bytes or more.

Reviewer: What happened when there were lines of  65,535 bytes or more?

Author: Worker panicked, don't need to call it out

This guide goes over how to upgrade to a specific Worker version and the updates for that version.

## Worker version 2.19.0

To upgrade to Worker version 2.19.0:

- Docker: Run the `docker pull` command for the [2.19.0 image][49].
- Kubernetes: See the [Helm chart][2] and [Upgrade the Worker][37].
- APT: Run the command `apt-get install observability-pipelines-worker=2.19.0`.
- RPM: Run the command `sudo yum install observability-pipelines-worker-2.19.0`.

Worker version 2.19.0 gives you access to the following:

#### New features

- The OpenTelemetry source and destination now support traces pipelines.
- The Custom Processor is now available in traces pipelines.
- The Sensitive Data Scanner processor is now available in traces pipelines.

> - Live Capture now supports traces pipelines.

Reviewer 2: Since trace pipelines are only available for demos, I think we don't want to advertise this all yet.

Author: Good point, will remove

#### Enhancements

- A `server_name` TLS option has been added to the BYOC Logs, HTTP Client, Socket, and Syslog destinations and to the HTTP Client source. It overrides the SNI and certificate hostname used for the TLS handshake, which is applicable when the dialed address does not match the certificate's Common Name or Subject Alternative Name.
- The Splunk TCP source now supports hot-reloading TLS certificates, so certificate rotations are applied without restarting the Worker.

> - The BYOC Logs destination now supports hot-reloading mTLS files.

Reviewer 2: Do we include the BYOC logs destination in the UI?

Author: Yup we do, that one is already GA
